tests/PHPStan/Rules/Methods/CallStaticMethodsRuleTest.php

```diff
@@ -432,17 +432,36 @@ public function testBug1971(): void
 	{
 		$this->checkThisOnly = false;
 		$this->analyse([__DIR__ . '/data/bug-1971.php'], [
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{\'Bug1971\\\HelloWorld\', \'sayHello\'} given.',
+				14,
+			],
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{class-string<static(Bug1971\\HelloWorld)>, \'sayHello\'} given.',
+				15,
+			],
 			[
 				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{class-string<static(Bug1971\\HelloWorld)>, \'sayHello2\'} given.',
 				16,
 			],
 		]);
 	}
 
-	public function testBug5259(): void
+	public function testBug727(): void
 	{
+		// \PHPStan\Type\Constant\ConstantArrayType::findTypeAndMethodName fails
+		// to find class in reflection since it has not been discovered, which
+		// prevents discovering the method properly. Only fix was to run a
+		// require on the file.
+		// Always was ConstantArrayTypeAndMethod::createUnknown()
+		require __DIR__ . '/data/bug-727.php';
 		$this->checkThisOnly = false;
-		$this->analyse([__DIR__ . '/data/bug-5259.php'], []);
+		$this->analyse([__DIR__ . '/data/bug-727.php'], [
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{class-string<static(Bug727\\HelloWorld)>, \'sayHello2\'} given.',
+				16,
+			],
```

Comment:
This was interesting. The `self::class` was failing since the class could not be autoloaded and the ConstantArrayType was always returning `ConstantArrayTypeAndMethod::createUnknown();`.

Running a `require` allowed the reflection provider to discover the class and report the expected errors.

Reply:
The require should not be necessary if you run `composer dump` before.